docs/tables/aws_s3_server_access_log/queries.md

```diff
@@ -0,0 +1,272 @@
+select
+  timestamp,
+  requester,
+  bucket,
+  key,
+  count(*) as access_count
+from
+  aws_s3_server_access_log
+group by
+  timestamp,
+  requester,
+  bucket,
+  key
+order by
+  access_count desc
+limit 10;
+```
+
+### Top 10 requesters
+
+Identify the top 10 requesters generating the most traffic.
+
+```sql
+select
+  timestamp,
+  requester,
+  count(*) as request_count
+from
+  aws_s3_server_access_log
+group by
+  timestamp,
+  requester
+order by
+  request_count desc
+limit 10;
+```
+
+### Top error codes
+
+Identify the most frequent error codes.
+
+```sql
+select
+  timestamp,
+  requester,
+  error_code,
+  count(*) as access_count
+from
+  aws_s3_server_access_log
+where
+  error_code is not null
+group by
+  timestamp,
+  requester,
+  error_code
+order by
+  access_count desc;
+```
+
+## Detection Examples
+
+### Unusual large file downloads
+
+Detect unusually large downloads from S3.
+
+```sql
+select
+  timestamp,
+  requester,
+  bucket,
+  key,
+  bytes_sent,
+  remote_ip,
+  user_agent
+from
+  aws_s3_server_access_log
+where
+  bytes_sent is not null
+  and bytes_sent > 50000000 -- 50MB
+order by
+  bytes_sent desc;
+```
+
+### Requests from unknown IPs
+
+Identify S3 access from unknown or unapproved IP addresses.
+
+```sql
+select
+  timestamp,
+  requester,
+  bucket,
+  remote_ip,
+  operation
+from
+  aws_s3_server_access_log
+where
+  remote_ip not in ('192.0.2.146', '206.253.208.100')
+order by
+  timestamp desc;
+```
+
+## Operational Examples
+
+### Find failed requests
+
+Identify all S3 requests that resulted in an error.
+
+```sql
+select
+  timestamp,
+  requester,
+  bucket,
+  request_uri,
+  http_status,
+  error_code,
+  remote_ip,
+  user_agent
+from
+  aws_s3_server_access_log
+where
+  http_status is not null
+  and http_status >= 400
+order by
+  timestamp desc;
+```
+
+### Detect ACL-related access issues
+
+Find failed requests where ACL permissions were required, helping identify potential permission misconfigurations or access control issues.
+
+```sql
+select
+  timestamp,
+  requester,
+  bucket,
+  key,
+  operation,
+  error_code,
+  http_status,
+  acl_required
+from
+  aws_s3_server_access_log
+where
+  acl_required = true
+  and (
+    http_status >= 400
+    or error_code is not null
+  )
+order by
+  timestamp desc;
+```
+
+## Volume Examples
+
+### High volume of access requests
+
+Detect unusually high access activity to S3 buckets and objects.
+
+```sql
+select
+  timestamp,
+  requester,
+  bucket,
+  count(*) as access_count,
+  date_trunc('minute', timestamp) as access_minute
+from
+  aws_s3_server_access_log
+group by
+  timestamp,
+  requester,
+  bucket,
+  access_minute
+having
+  count(*) > 100
+order by
+  access_count desc;
+```
+
+### High volume of failed requests
+
+Identify accounts with a high number of failed requests.
+
+```sql
+select
+  timestamp,
+  requester,
+  count(*) as failed_requests
+from
+  aws_s3_server_access_log
+where
+  http_status is not null
```

Comment:
Is it HTTP status codes > 400 or where there's any error code?

Reply:
It should be `status codes > 400 ` only